Client/Ecosia/Helpers/EcosiaInstallType/EcosiaInstallType+Extensions.swift

```diff
@@ -26,6 +27,7 @@ extension EcosiaInstallType {
         
         if EcosiaInstallType.persistedCurrentVersion() != versionProvider.version {
             EcosiaInstallType.set(type: .upgrade)
+            guard storeUpgradeVersion else { return }
             EcosiaInstallType.updateCurrentVersion(version: versionProvider.version)
```

Comment:
In which use case would we want to change the install type but not the version stored? To me those seem like they should always be side to side, let me know if I'm missing something.

I guess updating this on AppDelegate might be too soon if the user very quickly dismisses the app without seeing the popup? If so can't we remove from AppDelegate? Is there any other place in between this and the coordinator where install type is used?

Reply:
Hey @lucaschifino , thanks for your comment!
The main reason to have it at that point is due to the inner check we do with the `WhatsNewLocalDataProvider.shouldShowWhatsNewPage`.
We check whether `EcosiaInstallType.get() == .upgrade` hence we need to have it marked as `upgrade` already. However, to now fail and not show the WhatsNewPage we need to store the new version only after seeing the popup.
I'm conscious this creates a dependency with the Business Logic at the moment as the WhatsNewPage needs to know the upgrading version prior to the `EcosiaInstallType` storing this.